console/console-init/ui/src/modules/iot-device-detail/dialogs/CloneDevice/CloneDevicePage.tsx

```diff
@@ -64,46 +99,191 @@ export default function CloneDevicePage() {
     ),
     [projectname]
   );
-
+  const onDeviceSaveSuccess = () => {
+    resetWizardState();
+    history.push(deviceListRouteLink);
+  };
+  const resetWizardState = () => {
+    setDeviceIdInput("");
+    setIsEnabled(false);
+    setGatewayDevices([]);
+    setGatewayGroups([]);
+    setMemberOf([]);
+    setCredentials([getCredentialsFieldsInitialState()]);
+    setMetadataList(getInitalMetaData());
+  };
+  const [setCreateDeviceQueryVariables] = useMutationQuery(
+    CREATE_IOT_DEVICE,
+    undefined,
+    undefined,
+    onDeviceSaveSuccess
+  );
   useBreadcrumb(breadcrumb);
+  const queryResolver = `
+devices{
+registration{
+enabled
+via
+memberOf
+viaGroups
+ext
+defaults
+}
+}
+`;
+
+  const { loading, data } = useQuery<IDeviceDetailResponse>(
```

Comment:
Loading is unused here, if we need it, then conditionally display the loading state else remove it.